### PR TITLE
Feature/secure ingress cognito

### PR DESCRIPTION
**Pull request by rameshv29**
*Issue #, if available:*

*Description of changes:*

Amazon EKS customers that use Application Load Balancer (ALB) can use Amazon Cognito to handle user registration and authentication without writing code to handle routine tasks such as user sign-up, sign-in, sign-out, and so on. This is because Application Load Balancer has built-in support for user authentication. In addition to Amazon Cognito, ALB natively integrates with any OpenID Connect protocol compliant identity provider (IdP), providing secure authentication and a single sign-on experience across your applications.

In this pattern, we will be demonstrating the process to setup Kubecost on Amazon EKS and authentication users to Kubecost on Amazon EKS using CDK EKS Blueprints patterns.  The Kubecost dashboard does not provide native support for authentication. It requires an external authentication mechanism. This pattern will help customers secure their application's ingress using Amazon Cognito authentication. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/secure-ingress-auth-cognito/index.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+import { LookupHostedZoneProvider, GlobalResources, utils, ClusterInfo } from '@aws-quickstart/eks-blueprints';
+import { KubecostAddOn } from '@kubecost/kubecost-eks-blueprints-addon';
+import * as cdk from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { prevalidateSecrets } from '../common/construct-utils';
+import { SECRET_ARGO_ADMIN_PWD } from '../multi-region-construct';
+import * as cognito from 'aws-cdk-lib/aws-cognito';
+
+
+const logger = blueprints.utils.logger;
```
Comment (shapirov103) [inline lib/secure-ingress-auth-cognito/index.ts:12]: Please address warning that logger is assigned a value but never used (i.e. drop the variable).

File: lib/secure-ingress-auth-cognito/index.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                    required: true
+                }
+            }
+        });
+        
+        // Output the User Pool ID
+
+        this.userPoolOut = userPool;
+
+        const userPoolOut = new cdk.CfnOutput(this, 'CognitoIDPUserPoolOut', {
```
Comment (shapirov103) [inline lib/secure-ingress-auth-cognito/index.ts:58]: GitHub action gave a warning on this and next variable: usePoolOut is assigned a value but never used. 
This is because CDK is a weird library from the typescript perspective as they  use constructor where a method should have been used. 

Can you please add `// @ts-ignore` comment above each such case (there are number of such) to suppress the warning?

File: lib/secure-ingress-auth-cognito/index.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            value: userPool.userPoolId,
+            exportName: 'CognitoIDPUserPoolId'
+        });
+
+        const userPoolArnOut = new cdk.CfnOutput(this, 'CognitoIDPUserPoolArnOut', {
```
Comment (shapirov103) [inline lib/secure-ingress-auth-cognito/index.ts:63]: `userPoolArnOut` is assigned a value but never usrn

File: lib/secure-ingress-auth-cognito/index.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            },
+            generateSecret: true,
+            userPoolClientName: 'Web',
+            supportedIdentityProviders: [cognito.UserPoolClientIdentityProvider.COGNITO]
+        });
+
+        // Output the User Pool App Client ID
+        this.userPoolClientOut = userPoolClient;
+
+        const userPoolClientOut = new cdk.CfnOutput(this, 'CognitoIDPUserPoolClientOut', {
```
Comment (shapirov103) [inline lib/secure-ingress-auth-cognito/index.ts:96]: value not used warning.

File: lib/secure-ingress-auth-cognito/index.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        const userPoolDomain = userPool.addDomain('CognitoDomain', {
+            cognitoDomain: {
+                domainPrefix: 'my-cdk-blueprint',
+            },
+        });
+
+        // Output the User Pool App Client ID
+
+        this.userPoolDomainOut = userPoolDomain;
+        const useruserPoolDomainOut = new cdk.CfnOutput(this, 'CognitoIDPUserPoolDomainOut', {
```
Comment (shapirov103) [inline lib/secure-ingress-auth-cognito/index.ts:111]: value not used warning.

File: docs/patterns/secureingresscognito.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
+# Secure Ingress using Cognito Pattern
+
+## Objective
+
+Amazon EKS customers that use Application Load Balancer (ALB) can use Amazon Cognito to handle user registration and authentication without writing code to handle routine tasks such as user sign-up, sign-in, sign-out, and so on. This is because Application Load Balancer has built-in support for user authentication. In addition to Amazon Cognito, ALB natively integrates with any OpenID Connect protocol compliant identity provider (IdP), providing secure authentication and a single sign-on experience across your applications.
```
Comment (shapirov103) [inline docs/patterns/secureingresscognito.md:5]: customers <s>that</s> who

File: docs/patterns/secureingresscognito.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+In this pattern, we will be demonstrating the process to setup Kubecost on Amazon EKS and authentication users to Kubecost on Amazon EKS using CDK EKS Blueprints patterns.  The Kubecost dashboard does not provide native support for authentication. It requires an external authentication mechanism. This pattern will help customers secure their application's ingress using Amazon Cognito authentication.
```
Comment (shapirov103) [inline docs/patterns/secureingresscognito.md:7]: Minor: we could call out that "while the pattern is based on Kubecost, the approach can be reused with any similar products or with customer applications".

File: docs/patterns/secureingresscognito.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+secure-ingress-blueprint.secureingressblueprintConfigCommandD0275968 =  aws eks update-kubeconfig —name secure-ingress-blueprint —region us-west-2 —role-arn arn:aws:iam::<ACCOUNT ID>:role/secure-ingress-blueprint-secureingressblueprintMas-7JD5S67SG7M0
+secure-ingress-blueprint.secureingressblueprintGetTokenCommand21BE2184 =  aws eks get-token —cluster-name secure-ingress-blueprint —region us-west-2 —role-arn arn:aws:iam::<ACCOUNT ID>:role/secure-ingress-blueprint-secureingressblueprintMas-7JD5S67SG7M0
+```
+```
+Stack ARN:
+arn:aws:cloudformation:us-west-2:<ACCOUNT ID>:stack/secure-ingress-blueprint/64017120-91ce-11ed-93b2-0a67951f5d5d
+```
+
+
+To update your Kubernetes config for you new cluster, copy and run the secure-ingress-blueprint.secureingressblueprintConfigCommandD0275968 command (the second command) in your terminal.
```
Comment (shapirov103) [inline docs/patterns/secureingresscognito.md:96]: for <s>you</s> your new cluster

File: docs/patterns/secureingresscognito.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+```
+kubectl get namespace
+```
+
+You should see output that lists all namespaces in your cluster.
+
+
+## Test authentication
+
+Point your browsers to the URL of the Kubecost app in your cluster. Your browser will be redirected to a sign-in page. This page is provided by Amazon Cognito hosted UI.
```
Comment (shapirov103) [inline docs/patterns/secureingresscognito.md:113]: browser (singular)

File: lib/secure-ingress-auth-cognito/index.ts [hunk 10]
```diff
@@ -0,0 +1,186 @@
+import 'source-map-support/register';
+import * as blueprints from '@aws-quickstart/eks-blueprints';
+import { LookupHostedZoneProvider, GlobalResources, utils, ClusterInfo } from '@aws-quickstart/eks-blueprints';
```
Comment (shapirov103) [inline lib/secure-ingress-auth-cognito/index.ts:3]: Please removed unused imports for LookupHostedZoneProvider, ClusterInfo

File: README.md [hunk 11]
```diff
@@ -13,10 +13,31 @@ The individual patterns can be found in the `lib` directory.  Most of the patter
 Please refer to the Amazon EKS Blueprints Quick Start [documentation site](https://aws-quickstart.github.io/cdk-eks-blueprints/) for complete project documentation.
```
Comment (shapirov103) [inline README.md:13]: Thank you for making it better and fixing it, much appreciated!

**Review by shapirov103 — COMMENTED** [review]
@rameshv29 looks great, and thank you for fixing the readme along with it. 
I added a few minor comments, when you get a chance. 
I will merge once you apply the changes in the workloads repo (slight refactoring will be needed here as well).

File: docs/patterns/secureingresscognito.md (hunk 7 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:7]: thank you. updated.

File: docs/patterns/secureingresscognito.md (hunk 8 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:96]: thank you. updated.

File: docs/patterns/secureingresscognito.md (hunk 9 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:113]: thank you. updated.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 10 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:3]: thank you ! updated.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 1 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:12]: thank you ! updated.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 2 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:58]: thank you ! Updated // @ts-ignore for all such occurrences.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 4 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:96]: thank you ! Updated // @ts-ignore for all such occurrences.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 5 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:111]: thank you ! Updated // @ts-ignore for all such occurrences.

File: README.md (hunk 11 above)
Comment (rameshv29) [reply README.md:13]: Thank you very much @shapirov103 !! Happy to be part of this team !!

**Comment by rameshv29** [comment]
thanks @shapirov103 !! updated all the review comments and pushed the PR.

**Comment by rameshv29** [comment]
removed ts-ignore since lint throwing errors and updated the code.

File: cdk.json [hunk 12]
```diff
@@ -1,4 +1,8 @@
 {
```
Comment (elamaran11) [inline cdk.json:1]: This file should not be checked in but you should rather add this in readMe documentation of the pattern to show how to create a `cdk.json` file and how to use it before `cdk deploy`

File: lib/secure-ingress-auth-cognito/index.ts [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+import { prevalidateSecrets } from '../common/construct-utils';
+import { SECRET_ARGO_ADMIN_PWD } from '../multi-region-construct';
+import * as cognito from 'aws-cdk-lib/aws-cognito';
+
+const gitUrl = 'https://github.com/aws-samples/eks-blueprints-workloads.git';
+
+//Class Cognito Stack 
+
+/**
+ * Stack the creates the cognito user pool, app client, configure the client and app client domain. .
```
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:16]: There is an additional . at the end. COuld you please add more narrative of what exactly the stack is doing.

File: lib/secure-ingress-auth-cognito/index.ts [hunk 14]
```diff
@@ -0,0 +1,185 @@
...
+                                cognitoUserPoolArn: CognitoIdpStackOut.userPoolOut.userPoolArn,
+                                cognitoUserPoolAppId: CognitoIdpStackOut.userPoolClientOut.userPoolClientId,
+                                cognitoDomainName: CognitoIdpStackOut.userPoolDomainOut.domainName,
+                                region: process.env.CDK_DEFAULT_REGION,
+                            }
+                        },
+                    },
+                    adminPasswordSecretName: SECRET_ARGO_ADMIN_PWD,
+                }),
+                new blueprints.MetricsServerAddOn,
```
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:177]: Why do we need `MetricsServerAddon`  to this pattern?

File: lib/secure-ingress-auth-cognito/index.ts [hunk 15]
```diff
@@ -0,0 +1,185 @@
...
+            .region(process.env.CDK_DEFAULT_REGION)
+            .resourceProvider(GlobalResources.HostedZone, new LookupHostedZoneProvider(parentDomain))
+            .resourceProvider(GlobalResources.Certificate, new blueprints.CreateCertificateProvider('secure-ingress-cert', `${subdomain}`, GlobalResources.HostedZone))
+            .addOns(
+                new blueprints.VpcCniAddOn(),
+                new blueprints.CoreDnsAddOn(),
+                new blueprints.CertManagerAddOn,
+                new blueprints.AwsLoadBalancerControllerAddOn,
+                new KubecostAddOn({kubecostToken: "cmVhY2hyaytrdWJlY29zdEBhbWF6b24uY29txm343yadf98"}),
+                new blueprints.addons.EbsCsiDriverAddOn(),
```
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:153]: Same why do we need EBS Addon?

File: lib/secure-ingress-auth-cognito/index.ts [hunk 16]
```diff
@@ -0,0 +1,185 @@
...
+
+        await blueprints.EksBlueprint.builder()
+            .account(process.env.CDK_DEFAULT_ACCOUNT)
+            .region(process.env.CDK_DEFAULT_REGION)
+            .resourceProvider(GlobalResources.HostedZone, new LookupHostedZoneProvider(parentDomain))
+            .resourceProvider(GlobalResources.Certificate, new blueprints.CreateCertificateProvider('secure-ingress-cert', `${subdomain}`, GlobalResources.HostedZone))
+            .addOns(
+                new blueprints.VpcCniAddOn(),
+                new blueprints.CoreDnsAddOn(),
+                new blueprints.CertManagerAddOn,
```
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:150]: Same why do we need this. I would assume certificate to be managed by ACM in this pattern.

File: docs/patterns/secureingresscognito.md [hunk 17]
```diff
@@ -0,0 +1,132 @@
+# Secure Ingress using Cognito Pattern
+
+## Objective
+
+Amazon EKS customer who use Application Load Balancer (ALB) can use Amazon Cognito to handle user registration and authentication without writing code to handle routine tasks such as user sign-up, sign-in, sign-out, and so on. This is because Application Load Balancer has built-in support for user authentication. In addition to Amazon Cognito, ALB natively integrates with any OpenID Connect protocol compliant identity provider (IdP), providing secure authentication and a single sign-on experience across your applications.
```
Comment (elamaran11) [inline docs/patterns/secureingresscognito.md:5]: Narrative is not clearly explaining our intent and architecture here. Also missing the role of ACM and Route53 along with Integrating with Cognito.

File: docs/patterns/secureingresscognito.md [hunk 18]
```diff
@@ -0,0 +1,132 @@
...
+![Kubecost Architecture](./images/secure-ingress-kubecost.png)
+
+
+## Approach
+
+This blueprint will include the following:
+
+* A new Well-Architected VPC with both Public and Private subnets.
+* A new Well-Architected EKS cluster in the region and account you specify.
+* [Metrics Server](https://github.com/kubernetes-sigs/metrics-server) into your cluster to support metrics collection.
```
Comment (elamaran11) [inline docs/patterns/secureingresscognito.md:20]: Is this needed?

File: docs/patterns/secureingresscognito.md [hunk 19]
```diff
@@ -0,0 +1,132 @@
+# Secure Ingress using Cognito Pattern
+
+## Objective
+
+Amazon EKS customer who use Application Load Balancer (ALB) can use Amazon Cognito to handle user registration and authentication without writing code to handle routine tasks such as user sign-up, sign-in, sign-out, and so on. This is because Application Load Balancer has built-in support for user authentication. In addition to Amazon Cognito, ALB natively integrates with any OpenID Connect protocol compliant identity provider (IdP), providing secure authentication and a single sign-on experience across your applications.
+
+While the pattern is based on Kubecost, the approach can be reused with any similar products or with customer applications. The Kubecost dashboard does not provide native support for authentication. It requires an external authentication mechanism. This pattern will help customers secure their application's ingress using Amazon Cognito authentication.
```
Comment (elamaran11) [inline docs/patterns/secureingresscognito.md:7]: Talk couple of lines about KubeCost, then what are we doing with this pattern with Kubecost. Also then explain how this can be extended for apps

File: docs/patterns/secureingresscognito.md [hunk 20]
```diff
@@ -0,0 +1,132 @@
...
+
+## Approach
+
+This blueprint will include the following:
+
+* A new Well-Architected VPC with both Public and Private subnets.
+* A new Well-Architected EKS cluster in the region and account you specify.
+* [Metrics Server](https://github.com/kubernetes-sigs/metrics-server) into your cluster to support metrics collection.
+* AWS and Kubernetes resources needed to support [Cluster Autoscaler](https://docs.aws.amazon.com/eks/latest/userguide/cluster-autoscaler.html).
+* [EBS CSI Driver Amazon EKS Add-on](https://aws-quickstart.github.io/cdk-eks-blueprints/addons/ebs-csi-driver/) allows Amazon Elastic Kubernetes Service (Amazon EKS) clusters to manage the lifecycle of Amazon EBS volumes for persistent volumes.
```
Comment (elamaran11) [inline docs/patterns/secureingresscognito.md:22]: THis is also need IMO

File: docs/patterns/secureingresscognito.md [hunk 21]
```diff
@@ -0,0 +1,132 @@
...
+* [Argo CD](https://aws-quickstart.github.io/cdk-eks-blueprints/addons/argo-cd/) is a declarative, GitOps continuous delivery tool for Kubernetes. The Argo CD add-on provisions Argo CD into an EKS cluster, and bootstraping your workloads from public and private Git repositories.
+* Create the necessary Cognito resources like user pool, user pool client, domain etc.., and passed to the Argo CD app of apps pattern from which ingress resources can reference.
+
+## GitOps confguration
+
+For GitOps, the blueprint bootstrap the ArgoCD addon and points to the [EKS Blueprints Workload](https://github.com/aws-samples/eks-blueprints-workloads) sample repository.
+
+
+## Prerequisites
+1. `argo-admin-password` secret must be defined as plain text (not key/value) in `us-west-2`  region.
```
Comment (elamaran11) [inline docs/patterns/secureingresscognito.md:36]: If this handled in deployment section, we should remove this from here

File: docs/patterns/secureingresscognito.md [hunk 22]
```diff
@@ -0,0 +1,132 @@
...
+2. The parent domain must exist.
+3. The actual settings for the parent domain, hosted zone name and expected subzone name are expected to be specified in the CDK context. Generically it is inside the cdk.context.json file of the current directory or in `~/.cdk.json` in your home directory. Example settings:
```
Comment (elamaran11) [inline docs/patterns/secureingresscognito.md:38]: Same comment.

File: docs/patterns/secureingresscognito.md [hunk 23]
```diff
@@ -0,0 +1,132 @@
...
+To update your Kubernetes config for your new cluster, copy and run the secure-ingress-blueprint.secureingressblueprintConfigCommandD0275968 command (the second command) in your terminal.
+
+```
+aws eks update-kubeconfig —name secure-ingress-blueprint —region us-west-2 —role-arn arn:aws:iam::<ACCOUNT ID>:role/secure-ingress-blueprint-secureingressblueprintMas-7JD5S67SG7M0
+```
+
+Validate that you now have kubectl access to your cluster via the following:
+
+```
+kubectl get namespace
```
Comment (elamaran11) [inline docs/patterns/secureingresscognito.md:105]: Show all Kubecost k8s resources.

File: docs/patterns/secureingresscognito.md [hunk 24]
```diff
@@ -0,0 +1,132 @@
...
+```
+
+You should see output that lists all namespaces in your cluster.
+
+
+## Test authentication
+
+Point your browser to the URL of the Kubecost app in your cluster. Your browser will be redirected to a sign-in page. This page is provided by Amazon Cognito hosted UI.
```
Comment (elamaran11) [inline docs/patterns/secureingresscognito.md:113]: Can you show this in automation. You can follow our Cognito, Yelb blog for this

File: lib/secure-ingress-auth-cognito/index.ts (hunk 14 above)
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:177]: Why is this here?

File: lib/secure-ingress-auth-cognito/index.ts [hunk 25]
```diff
@@ -0,0 +1,185 @@
...
+                                cognitoUserPoolAppId: CognitoIdpStackOut.userPoolClientOut.userPoolClientId,
+                                cognitoDomainName: CognitoIdpStackOut.userPoolDomainOut.domainName,
+                                region: process.env.CDK_DEFAULT_REGION,
+                            }
+                        },
+                    },
+                    adminPasswordSecretName: SECRET_ARGO_ADMIN_PWD,
+                }),
+                new blueprints.MetricsServerAddOn,
+                new blueprints.ClusterAutoScalerAddOn)
```
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:178]: Why is this needed for this patern?

**Review by elamaran11 — COMMENTED** [review]
@rameshv29 Great work with the PR. Have some comments for you to address, Please let me know if you have any questions.

File: cdk.json (hunk 12 above)
Comment (rameshv29) [reply cdk.json:1]: updated the git ignore and instructions were already added in read me file.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 13 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:16]: Updated.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 14 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:177]: updated the code.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 16 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:150]: updated the code.

File: docs/patterns/secureingresscognito.md (hunk 23 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:105]: updated

File: lib/secure-ingress-auth-cognito/index.ts (hunk 14 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:177]: updated the code.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 25 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:178]: updated the code

File: docs/patterns/secureingresscognito.md (hunk 18 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:20]: updated

File: docs/patterns/secureingresscognito.md (hunk 20 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:22]: updated

File: docs/patterns/secureingresscognito.md (hunk 21 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:36]: updated

File: docs/patterns/secureingresscognito.md (hunk 22 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:38]: updated

File: docs/patterns/secureingresscognito.md (hunk 17 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:5]: updated

File: docs/patterns/secureingresscognito.md (hunk 19 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:7]: updated

File: docs/patterns/secureingresscognito.md (hunk 24 above)
Comment (rameshv29) [reply docs/patterns/secureingresscognito.md:113]: updated.

**Comment by rameshv29** [comment]
updated the code and read me file based on previous comments and pushed.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 16 above)
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:150]: I still see `CertManagerAddOn` in the code, But the comment is resolved.

File: lib/secure-ingress-auth-cognito/index.ts [hunk 26]
```diff
@@ -0,0 +1,185 @@
...
+            .account(process.env.CDK_DEFAULT_ACCOUNT)
+            .region(process.env.CDK_DEFAULT_REGION)
+            .resourceProvider(GlobalResources.HostedZone, new LookupHostedZoneProvider(parentDomain))
+            .resourceProvider(GlobalResources.Certificate, new blueprints.CreateCertificateProvider('secure-ingress-cert', `${subdomain}`, GlobalResources.HostedZone))
+            .addOns(
+                new blueprints.VpcCniAddOn(),
+                new blueprints.CoreDnsAddOn(),
+                new blueprints.CertManagerAddOn,
+                new blueprints.AwsLoadBalancerControllerAddOn,
+                new KubecostAddOn({kubecostToken: "cmVhY2hyaytrdWJlY29zdEBhbWF6b24uY29txm343yadf98"}),
```
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:152]: Why is the kubecostToken hard coded, you should use secrets manager to pull it like you are doing for others.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 14 above)
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:177]: Same, you have resolved the comment but i still see `MetricsServerAddOn`

File: lib/secure-ingress-auth-cognito/index.ts (hunk 25 above)
Comment (elamaran11) [inline lib/secure-ingress-auth-cognito/index.ts:178]: Same with `ClusterAutoScalerAddOn`

File: docs/patterns/secureingresscognito.md (hunk 18 above)
Comment (elamaran11) [inline docs/patterns/secureingresscognito.md:20]: This is till not removed from the comment.

File: docs/patterns/secureingresscognito.md (hunk 21 above)
Comment (elamaran11) [reply docs/patterns/secureingresscognito.md:36]: @rameshv29 I dont see you have handled this comment but the comment is resolved.

File: docs/patterns/secureingresscognito.md (hunk 22 above)
Comment (elamaran11) [reply docs/patterns/secureingresscognito.md:38]: @rameshv29 I dont see you have handled this comment but the comment is resolved.

File: docs/patterns/secureingresscognito.md (hunk 23 above)
Comment (elamaran11) [reply docs/patterns/secureingresscognito.md:105]: @rameshv29 I dont see you have handled this comment but the comment is resolved.

File: docs/patterns/secureingresscognito.md (hunk 24 above)
Comment (elamaran11) [reply docs/patterns/secureingresscognito.md:113]: @rameshv29 I dont see you have handled this comment but the comment is resolved.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 13 above)
Comment (elamaran11) [reply lib/secure-ingress-auth-cognito/index.ts:16]: @rameshv29 I dont see you have handled this comment but the comment is resolved.

**Review by elamaran11 — COMMENTED** [review]
@rameshv29 I see you have not incorporated the comments but the comments are resolved. I think you have not pushed the latest code.

File: lib/secure-ingress-auth-cognito/index.ts (hunk 26 above)
Comment (rameshv29) [reply lib/secure-ingress-auth-cognito/index.ts:152]: updated,

File: lib/secure-ingress-auth-cognito/index.ts (hunk 13 above)
Comment (shapirov103) [reply lib/secure-ingress-auth-cognito/index.ts]: @rameshv29 @elamaran11 - making sure we are on the same page-  we use the following rules for code review:

1. Reviewer adds comments, and only reviewer (not committer) marks their comment as resolved.
2. Committer addresses the feedback and replies to the comments with `Addressed`.
3. If reviewer agrees that the comment is addressed, they mark it as `resolved` or they approve the PR as a whole, meaning all feedback is addressed.

**Review by shapirov103 — APPROVED** [review]
LGTM

**Comment by shapirov103** [comment]
@elamaran11 please review your comments and merge once you approve this.

**Review by elamaran11 — APPROVED** [review]
LGTM @rameshv29 @shapirov103
